### PR TITLE
dune3d: init at 1.0.0

### PR DESCRIPTION
**Pull request by 0x4A6F**
## Description of changes

Add [dune3d](https://github.com/dune3d/dune3d) to nixpkgs.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/du/dune3d/package.nix
Comment (carrotIndustries) [inline pkgs/by-name/du/dune3d/package.nix:59]: https://dune3d.org/ is the proper homepage

File: pkgs/by-name/du/dune3d/package.nix
Comment (jue89) [inline pkgs/by-name/du/dune3d/package.nix:55]:
```suggestion
  env.CASROOT = opencascade-occt;
```

File: pkgs/by-name/du/dune3d/package.nix
Comment (jue89) [inline pkgs/by-name/du/dune3d/package.nix:60]:
```suggestion
    license = licenses.gpl3Plus;
```

cf. https://github.com/dune3d/dune3d/blob/58613f1305931523b58b60d94f25668ef5392d1d/org.dune3d.dune3d.metainfo.xml#L6

File: pkgs/by-name/du/dune3d/package.nix
Comment (jue89) [inline pkgs/by-name/du/dune3d/package.nix:63]:
```suggestion
    platforms = platforms.linux;
```

Or does it build for macOS? IIRC the macOS build heavily depends on `brew`.

File: pkgs/by-name/du/dune3d/package.nix
Comment (jue89) [inline pkgs/by-name/du/dune3d/package.nix:61]: If you want to, you could add me as a maintainer :)
I'm already in the maintainers list: `jue89`.

**Review by jue89 — CHANGES_REQUESTED** [review]
Thanks for adding Dune3D to nixpkgs!

I haven't tested your PR, yet. Here's my first round of comments ...

File: pkgs/by-name/du/dune3d/package.nix
Comment (0x4A6F) [reply pkgs/by-name/du/dune3d/package.nix]: Thanks, you are welcome.

**Comment by jue89** [comment]
Dune3D builds without any problems:

```
jue@baxxter ~/P/nixpkgs (master)> nix run nixpkgs#nixpkgs-review pr 276931
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/276931/head:refs/nixpkgs-review/1
remote: Enumerating objects: 250738, done.
remote: Counting objects: 100% (43586/43586), done.
remote: Compressing objects: 100% (290/290), done.
remote: Total 250738 (delta 43429), reused 43397 (delta 43293), pack-reused 207152
Receiving objects: 100% (250738/250738), 170.81 MiB | 1.10 MiB/s, done.
Resolving deltas: 100% (170045/170045), completed with 9306 local objects.
From https://github.com/NixOS/nixpkgs
   91d118af62b1..0e148322b344  master                -> refs/nixpkgs-review/0
 + 8f5532ae2b89...3315bf8fbb80 refs/pull/276931/head -> refs/nixpkgs-review/1  (forced update)
$ git worktree add /home/jue/.cache/nixpkgs-review/pr-276931/nixpkgs 0e148322b344eab7c8d52f6e59b0d95ba73fb62e
Preparing worktree (detached HEAD 0e148322b344)
Updating files: 100% (39206/39206), done.
HEAD is now at 0e148322b344 Merge pull request #283494 from andresilva/polkadot-v1.6.0
$ git merge --no-commit --no-ff 3315bf8fbb801019df6e44e3f94971835e412d1c
Automatic merge went well; stopped before committing as requested
$ nix build --nix-path nixpkgs=/home/jue/.cache/nixpkgs-review/pr-276931/nixpkgs nixpkgs-overlays=/tmp/tmpbjcbtu3r --extra-experimental-features nix-command no-url-literals --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed -f /home/jue/.cache/nixpkgs-review/pr-276931/build.nix

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/276931

1 package built:
dune3d
```

File: pkgs/by-name/du/dune3d/package.nix
Comment (jue89) [inline pkgs/by-name/du/dune3d/package.nix:60]:
```suggestion
    license = licenses.gpl3Plus;
```

Indentation went wrong ;-)

**Review by jue89 — CHANGES_REQUESTED** [review]
Last nit-pick from my side. Otherwise, this looks good to me!

But I'd say that we could wait for 1.0 being release the upcoming weekend (if I got @carrotIndustries right) before merging?

File: pkgs/by-name/du/dune3d/package.nix
Comment (0x4A6F) [reply pkgs/by-name/du/dune3d/package.nix]: Switching from work-layout (de) to home-layout (us) hiccup, sorry.

**Review by jue89 — APPROVED** [review]
Sweet. LGTM!

**Comment by jue89** [comment]
Still builds on latest master.

```
jue@baxxter ~/P/nixpkgs (master)> nix run 'nixpkgs#nixpkgs-review' pr 276931
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/276931/head:refs/nixpkgs-review/1
remote: Enumerating objects: 50694, done.
remote: Counting objects: 100% (15884/15884), done.
remote: Compressing objects: 100% (261/261), done.
remote: Total 50694 (delta 15712), reused 15729 (delta 15609), pack-reused 34810
Receiving objects: 100% (50694/50694), 29.67 MiB | 9.86 MiB/s, done.
Resolving deltas: 100% (35975/35975), completed with 4730 local objects.
From https://github.com/NixOS/nixpkgs
   0e148322b344..9002908215ae  master                -> refs/nixpkgs-review/0
 + 3315bf8fbb80...85fe7c5226ca refs/pull/276931/head -> refs/nixpkgs-review/1  (forced update)
$ git worktree add /home/jue/.cache/nixpkgs-review/pr-276931-1/nixpkgs 9002908215ae609210c2168494b387ff5728220a
Preparing worktree (detached HEAD 9002908215ae)
Updating files: 100% (39442/39442), done.
HEAD is now at 9002908215ae Merge pull request #285047 from Aleksanaa/tootle
$ git merge --no-commit --no-ff 85fe7c5226ca94d635596bf62f3c8cda70bfb12f
Automatic merge went well; stopped before committing as requested
$ nix build --nix-path nixpkgs=/home/jue/.cache/nixpkgs-review/pr-276931-1/nixpkgs nixpkgs-overlays=/tmp/tmp_d0fkj21 --extra-experimental-features nix-command no-url-literals --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed -f /home/jue/.cache/nixpkgs-review/pr-276931-1/build.nix

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/276931

1 package built:
dune3d
```

**Pull request merged**
